lib/logto_client.dart

```diff
@@ -43,43 +48,51 @@ class LogtoClient {
     }
 
     var discoveryUri = utils.appendUriPath(config.endpoint, discoveryPath);
-    _oidcConfig = await logto_core.fetchOidcConfig(httpClient, discoveryUri);
+    _oidcConfig = await logto_core.fetchOidcConfig(_httpClient, discoveryUri);
 
     return _oidcConfig!;
   }
 
-  // TODO: add loading to prevent multi click
-  Future signIn(
+  bool _loading = false;
+
+  Future<void> signIn(
     BuildContext context,
     String redirectUri,
     void Function() signInCallback,
   ) async {
+    if (_loading) return;
+    _loading = true;
     _pkce = PKCE.generate();
     _state = utils.generateRandomString();
-    _tokenStorage.idToken = null;
+    await _tokenStorage.setIdToken(null);
```

Comment:
maybe no need to await here? Clean up the token storage can run silently without blocking the main flow.

Reply:
yes, i guess, but you have to watch out that you don't run into issues with data inconsistency.